packages/tx-construction/test/tx-builder/TxBuilderDelegatePortfolio.test.ts

```diff
@@ -338,6 +358,20 @@ describe('TxBuilder/delegatePortfolio', () => {
       expect(roundRobinRandomImprove).toHaveBeenCalled();
     });
 
+    it('does nothing and uses random improve input selector when the wallet is multi delegating and transaction is not changing delegation', async () => {
+      const tx = await multiDelegatingWalletTxBuilder.build().inspect();
+      expect(tx.body.certificates).toBeUndefined();
+      expect(GreedyInputSelector).not.toHaveBeenCalled();
+      expect(roundRobinRandomImprove).toHaveBeenCalled();
+    });
+
+    it('does nothing and uses random improve input selector when the wallet is single delegating and transaction is not changing delegation', async () => {
+      const tx = await singleDelegatingWalletTxBuilder.build().inspect();
```

Comment:
Suggestion: you also include an output if the test fixtures allow it?
I see the input selection algorithm is invoked, so the test fulfils the purpose. Considering this, it is not mandatory to do this change, but it would make the test more meaningful, provided the fixtures support the change.

Reply:
Done here: [9a3d65d](https://github.com/input-output-hk/cardano-js-sdk/pull/1286/commits/9a3d65d52f12474365cb96aaf039155d11708142)